ports/esp8266/machine_pin.c

```diff
@@ -225,6 +215,16 @@ void pin_set(uint pin, int value) {
             break;
     }
 
+    if (pin == 16) {
+        WRITE_PERI_REG(PAD_XPD_DCDC_CONF, (READ_PERI_REG(PAD_XPD_DCDC_CONF) & 0xffffffbc) | 1);
+        WRITE_PERI_REG(RTC_GPIO_CONF, READ_PERI_REG(RTC_GPIO_CONF) & ~1);
+        WRITE_PERI_REG(RTC_GPIO_ENABLE, (READ_PERI_REG(RTC_GPIO_ENABLE) & ~1) | enable);
+        if (value != -1) {
+            WRITE_PERI_REG(RTC_GPIO_OUT, (READ_PERI_REG(RTC_GPIO_OUT) & ~1) | value);
+        }
+        return;
+    }
+
```

Comment:
This change looks good.  How did you test it?

Reply:
At software level by setting its value and checking that object returns exactly the same value that was set previously.
At hardware level first by a multimeter, then by actually controlling an active-low relay circuit.

Reply:
I don't think this new code will work in open-drain mode when `value==-1`: in such a case this code here won't be reached and the pin won't be configured at all.